lib/ng/edit/ogc/module.js

```diff
@@ -0,0 +1,460 @@
+            var tileUrlFunction = function(tileCoord, pixelRatio, projection) {
+              var zxy = tileCoord;
+              if (zxy[1] < 0 || zxy[2] < 0) {
+                return "";
+              }
+              return urls[Math.round(Math.random()*3)] + name + '/' +
+                  zxy[0].toString()+'/'+ zxy[1].toString() +'/'+
+                  zxy[2].toString() +'.png';
+            };
+            layer.setSource(new ol.source.TileImage({
+              crossOrigin: null,
+              attributions: [
+                new ol.Attribution({
+                  html: /^world/.test(name) ?
+                      "<a href='http://mapbox.com'>MapBox</a> | Some Data &copy; OSM CC-BY-SA | <a href='http://mapbox.com/tos'>Terms of Service</a>" :
+                      "<a href='http://mapbox.com'>MapBox</a> | <a href='http://mapbox.com/tos'>Terms of Service</a>"
+                })
+              ],
+              tileGrid: new ol.tilegrid.TileGrid({
+                origin: [-128 * 156543.03390625, -128 * 156543.03390625],
+                resolutions: [
+                    156543.03390625, 78271.516953125, 39135.7584765625,
+                    19567.87923828125, 9783.939619140625, 4891.9698095703125,
+                    2445.9849047851562, 1222.9924523925781, 611.4962261962891,
+                    305.74811309814453, 152.87405654907226, 76.43702827453613,
+                    38.218514137268066, 19.109257068634033, 9.554628534317017,
+                    4.777314267158508, 2.388657133579254, 1.194328566789627,
+                    0.5971642833948135
+                ]
+              }),
+              tileUrlFunction: tileUrlFunction
+            }));
+            return layer;
+          }
+        }
+      };
+    }); 
+
+    module.service('stLayerBuilder', function($q, stAnnotateLayer, stBaseLayerBuilder) {
+      return {
+        buildEditableLayer: function(data, map) {
+          var layer = new EditableStoryLayer(data);
+          var deferred = $q.defer();
+          var promises = [];
+          var needsCaps = !(data.latlonBBOX && data.times && data.bbox && data.resolutions);
+          if (needsCaps) {
+            promises.push(stAnnotateLayer.loadCapabilities(layer, map));
+          } else {
+            stAnnotateLayer.setWMSSource(layer);
+          }
+          var needsDFT = !data.attributes;
+          if (needsDFT) {
+            promises.push(stAnnotateLayer.describeFeatureType(layer));
+          }
+          if (data.type === 'VECTOR' && !data.features) {
+            promises.push(stAnnotateLayer.getFeatures(layer, map));
+          } else {
+            promises.push(stAnnotateLayer.getStyleName(layer));
+          }
+          $q.all(
+            promises
+          ).then(function() {
+            deferred.resolve(layer);
+          });
+          return deferred.promise;
+        }
+      };
+    });
+
+    module.service('stStoryMapBuilder', function(stLayerBuilder, stBaseLayerBuilder) {
```

Comment:
Agreed, the split still needs to be done, I wanted to do that when the viewer example is setup.